test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -463,21 +482,160 @@ contract LoansForkTest is LoansTestBase {
+        // Convert cash amount to expected underlying amount using oracle's conversion
+        uint expectedUnderlying = pair.oracle.convertToBaseAmount(totalAmountToSwap, currentPrice);
 
-        // no price change so underlying out should be underlying in minus slippage and minus late fees
-        uint minUnderlyingOut = underlyingAmount;
-        uint minUnderlyingOutWithSlippage = (minUnderlyingOut * (100 - slippage) / 100) - lateFee;
+        uint minUnderlyingOutWithSlippage = expectedUnderlying * (100 - slippage) / 100 - lateFee;
         uint underlyingOut = closeLoan(pair, user, loanId, minUnderlyingOutWithSlippage);
+
         assertGe(underlyingOut, minUnderlyingOutWithSlippage);
         // Verify balance changes
         assertEq(userCashBefore - pair.cashAsset.balanceOf(user), loanAmount);
         assertEq(pair.underlying.balanceOf(user) - userUnderlyingBefore, underlyingOut);
     }
 
+    // price movement settlement tests
+
+    function testSettlementPriceAboveCallStrike() public {
+        // Create provider offer & open loan
+        uint offerId = createProviderOffer(pair, callstrikeToUse, offerAmount, pair.durations[1]);
+        (uint loanId,, uint loanAmount) = openLoan(pair, user, underlyingAmount, 0.3e6, offerId);
+
+        ICollarTakerNFT.TakerPosition memory position = pair.takerNFT.getPosition(loanId);
+        skip(pair.durations[1] / 2);
+        // Move price above call strike using lib
+        PriceManipulationLib.movePriceUpPastCallStrike(
+            vm,
+            address(pair.swapperUniV3.uniV3SwapRouter()),
+            whale,
+            pair.cashAsset,
+            pair.underlying,
+            pair.oracle,
+            position.callStrikePercent,
+            POOL_FEE_TIER,
+            AMOUNT_FOR_CALL_STRIKE
+        );
+
+        skip(pair.durations[1] / 2);
+
+        (uint expectedTakerWithdrawal,) =
+            pair.takerNFT.previewSettlement(position, pair.oracle.currentPrice());
+
+        // Total cash = taker withdrawal + loan repayment
+        // Convert expected total cash to underlying at current price
+        uint expectedUnderlyingOut =
+            pair.oracle.convertToBaseAmount(expectedTakerWithdrawal + loanAmount, pair.oracle.currentPrice());
+
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+
+        // Close loan and settle
+        closeAndCheckLoan(loanId, loanAmount, loanAmount + expectedTakerWithdrawal, 0);
+
+        // Check provider's withdrawable amount
+        uint providerWithdrawable = position.providerNFT.getPosition(position.providerId).withdrawable;
+        assertEq(providerWithdrawable, 0); // everything to user
+
+        // Check user's underlying balance change against calculated expected amount
+        assertApproxEqAbs(
+            pair.underlying.balanceOf(user) - userUnderlyingBefore,
+            expectedUnderlyingOut,
+            expectedUnderlyingOut * slippage / 100 // within slippage% of expected
```

Comment:
slippage being a percentage, but not in BIPS is getting confusing + 100 magic number